### PR TITLE
[Dashboard]Use node id (node ip) to distinguish nodes log indexes

### LINKED ISSUE #30659: [Dashboard] Logs page links has issue if multiple ray nodes running on the same machine
### What happened + What you expected to happen

Logs page links has issue if multiple ray nodes running on the same machine.

e.g., if I starts 4 ray worker nodes on a machine, then, the dashboard logs page shows:
image

The 3 red lines links does not work.

### Versions / Dependencies

Ray master version.

### Reproduction script

N/A

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by MisterLin1995**
## Why are these changes needed?

When multiple ray nodes running on the same machine, the current dashboard log index makes it difficult to distinguish these nodes. Changes in this PR set the log index texts from "http://{node_ip}:{agent_port}/logs" to "{node_id} ({node_ip})".
## Related issue number
Closes #30659

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
@rkooo567 should the UX change be approved by the owner?

**Comment by rkooo567** [comment]
@jovany-wang cc @scottsun94. Can you guys also share a screenshot of new changes?

**Comment by rkooo567** [comment]
Also we need a test

**Comment by MisterLin1995** [comment]
> @jovany-wang cc @scottsun94. Can you guys also share a screenshot of new changes?

image

File: dashboard/client/src/service/log.ts
Comment (MisterLin1995) [reply dashboard/client/src/service/log.ts:27]: When access the logs index page, the front end will truncate the url link if it does not start with 'http'. This case helps to avoid url truncation like this:
image

**Comment by scottsun94** [comment]
> > @jovany-wang cc @scottsun94. Can you guys also share a screenshot of new changes?
> 
> image

Thanks! One minor issue:
Could we change`Please choose an url to see logs for that node` to `Select a node to view logs`?

Others changes LGTM.

**Comment by MisterLin1995** [comment]
> Select a node to view logs

done

**Comment by MisterLin1995** [comment]
> > > @jovany-wang cc @scottsun94. Can you guys also share a screenshot of new changes?
> > 
> > 
> > image
> 
> Thanks! One minor issue: Could we change`Please choose an url to see logs for that node` to `Select a node to view logs`?
> 
> Others changes LGTM.

done

File: dashboard/client/src/service/log.ts
Comment (rkooo567) [reply dashboard/client/src/service/log.ts:27]: oh I meant to update the code comment. sorry for confusion..

File: dashboard/modules/log/log_head.py
Comment (rkooo567) [inline dashboard/modules/log/log_head.py:15]:
```suggestion
    LOG_INDEX_TEXT_TEMPLATE = "Node ID: {node_id} (IP: {node_ip})"
```

**Review by rkooo567 — CHANGES_REQUESTED** [review]
One last comment! I will merge it after it is addressed. Thank you!

**Review by jovany-wang — APPROVED** [review]
LGTM.
CI failure should be unrelated. Maybe @rkooo567 could do a double check before merging.

File: dashboard/modules/log/log_head.py
Comment (MisterLin1995) [reply dashboard/modules/log/log_head.py]: done

**Comment by scottsun94** [comment]
cc: @alanwguo for awareness

**Comment by MisterLin1995** [comment]
@rkooo567 Please check new changes and merge the request.

**Review by rkooo567 — APPROVED** [review]

**Comment by jovany-wang** [comment]
Nice work! Thanks for the contribution.

**Pull request merged**
